transitions/index.js

```diff
@@ -73,24 +103,34 @@ const processChange = (change, callback) => {
  * Load transitions using `require` based on what is in AVAILABLE_TRANSITIONS
  * constant and what is enabled in the `transitions` property in the settings
  * data.  Log warnings on failure.
+ *
+ * Tests can diabled autoEnableSystemTransitions for more control.
  */
-const loadTransitions = () => {
+const loadTransitions = (autoEnableSystemTransitions = true) => {
+
   const self = module.exports;
-  const configuredTransitions = config.get('transitions') || [];
-  _.each(configuredTransitions, (conf, key) => {
-    if (!conf || conf.disable) {
-      return logger.warn(`transition ${key} is disabled`);
-    }
-    if (AVAILABLE_TRANSITIONS.indexOf(key) === -1) {
-      return logger.warn(`transition ${key} not available.`);
+  const transitionsConfig = config.get('transitions') || [];
+
+  // Load all system or configured transitions
+  AVAILABLE_TRANSITIONS.forEach(transition => {
+    const conf = transitionsConfig[transition];
+
+    const system =
+      autoEnableSystemTransitions && SYSTEM_TRANSITIONS.includes(transition);
+
+    const disabled = conf && conf.disable;
+
+    if ((system && disabled) || (!system && !conf || disabled)) {
+      return logger.warn(`transition ${transition} is disabled`);
```

Comment:
Is there a typo here? It looks like you're allowing disabling system transitions, which I don't think is what you intended? I don't think we should allow disabling.

This logic is really confusing but I think this boils down to `if(disabled)` which is why I think there's a typo. If it's not a typo then this needs to be made much easier to read.